src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -9,25 +9,43 @@ sealed partial class IdnMapping
         private unsafe string GetAsciiCore(char* unicode, int count)
         {
             uint flags = Flags;
-            CheckInvalidIdnCharacters(unicode, count, flags, "unicode");
+            CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
-            char[] buf = new char[count];
-
-            for (int attempts = 2; attempts > 0; attempts--)
+            const int StackAllocThreshold = 512;
+            if (count < StackAllocThreshold)
             {
-                int realLen = Interop.GlobalizationNative.ToAscii(flags, unicode, count, buf, buf.Length);
-
-                if (realLen == 0)
+                char* output = stackalloc char[count];
+                return GetAsciiCore(flags, unicode, count, output, count, reattempt: true);
+            }
+            else
+            {
+                char[] output = new char[count];
+                fixed (char* pOutput = output)
                 {
-                    break;
+                    return GetAsciiCore(flags, unicode, count, pOutput, count, reattempt: true);
                 }
+            }
+        }
+
+        private unsafe string GetAsciiCore(uint flags, char* unicode, int count, char* output, int outputLength, bool reattempt)
```

Comment:
Nit: could you move the ```flags``` parameter to be after the ```count``` parameter so that the overload is consistent in ordering with the previous one?  Same for the similar GetUnicodeCore case below.